File: backend/app/schemas/student_event.py
"""Pydantic schemas for the STUDENT side of event discovery.

This is a thin, read-only *adapter* over the one real event-style table in
this architecture -- `industry_workshops`
(database/migrations/024_industry_workshops.sql). There is no `events`
table and none is introduced: a student-facing "event" is just a
normalized view of one PUBLISHED industry workshop.

`industry_workshops` already carries the RLS policy
"Authenticated users can view published workshops"
(select, `status = 'PUBLISHED'`), so a STUDENT caller can already read
those rows through build_user_client(); this adapter relies on that
rather than widening any policy.

There is NO registration model anywhere in the repository
(024_industry_workshops.sql explicitly has "no application/registration
table"), so this adapter exposes none and invents none -- the student
frontend shows a truthful "registration isn't available yet" state.
Events are strictly read-only for students; nothing here can create,
update, or delete a workshop row.

The student-facing event id is the raw workshop UUID -- there is only one
source table, so no prefixing is needed (unlike student_opportunity's
internship_/job_ split).
"""

from pydantic import BaseModel

# database/migrations/024_industry_workshops.sql -- industry_workshops.work_mode CHECK.
StudentEventWorkMode = ("ONSITE", "REMOTE", "HYBRID")


class StudentEventOrganizer(BaseModel):
    """Company display info for the organising Industry account, read from
    `industry_profiles` (RLS: "Authenticated users can view industry
    profiles"). `company_name` can be null -- an Industry account can
    publish a workshop before filling in its company profile -- and no
    `profiles` column (full_name, email, ...) is ever exposed here; a
    student has no RLS path to it."""

    id: str
    company_name: str | None = None
    industry_sector: str | None = None
    logo_url: str | None = None


class StudentEventSummary(BaseModel):
    """One PUBLISHED industry workshop, normalized for the browse list."""

    id: str
    title: str
    description: str
    location: str | None = None
    work_mode: str | None = None  # ONSITE / REMOTE / HYBRID -- the "online indicator"
    start_date: str | None = None
    application_deadline: str | None = None
    duration_days: int | None = None
    organizer: StudentEventOrganizer | None = None
    created_at: str | None = None


class StudentEventDetail(StudentEventSummary):
    """One PUBLISHED industry workshop, normalized for the detail page."""

    capacity: int | None = None
    eligibility_criteria: str | None = None
    # Always False in this phase: there is no canonical registration
    # backend. Surfaced as a real field so the frontend renders an honest
    # state from the API rather than a hard-coded assumption.
    registration_available: bool = False


class StudentEventListResponse(BaseModel):
    events: list[StudentEventSummary]

File: frontend/types/student-event.ts
// Mirrors backend/app/schemas/student_event.py, which is itself a
// read-only adapter over the `industry_workshops` table
// (database/migrations/024_industry_workshops.sql). Keep in sync.
//
// A student-facing "event" is one PUBLISHED industry workshop. There is no
// `events` table and no registration/attendance model in this
// architecture, so this type has no registration fields beyond the
// `registration_available` flag (always false this phase) the backend
// sends so the UI can render an honest state.

export const STUDENT_EVENT_WORK_MODES = ["ONSITE", "REMOTE", "HYBRID"] as const;
export type StudentEventWorkMode = (typeof STUDENT_EVENT_WORK_MODES)[number];

export const STUDENT_EVENT_WORK_MODE_LABELS: Record<StudentEventWorkMode, string> = {
  ONSITE: "In person",
  REMOTE: "Online",
  HYBRID: "Hybrid",
};

export interface StudentEventOrganizer {
  id: string;
  company_name: string | null;
  industry_sector: string | null;
  logo_url: string | null;
}

export interface StudentEventSummary {
  id: string;
  title: string;
  description: string;
  location: string | null;
  work_mode: StudentEventWorkMode | null;
  start_date: string | null;
  application_deadline: string | null;
  duration_days: number | null;
  organizer: StudentEventOrganizer | null;
  created_at: string | null;
}

export interface StudentEventDetail extends StudentEventSummary {
  capacity: number | null;
  eligibility_criteria: string | null;
  registration_available: boolean;
}

export interface StudentEventListResponse {
  events: StudentEventSummary[];
}
